### ISSUE TITLE
Connection from LED

### ISSUE THREAD

**Issue by tinue** [diogeneyves/dji-naza-led#3]
Hello, according to the source code, you read the LED signal from A4/A5. On the wiring diagram, these pins are not shown/connected. Therefore my question: How exactly do you retrieve the basic color signal for the LED from the NAZA?

**Comment by diogeneyves** [comment]
hello,

You're right, it is not correct with the schema. I modified the source code accordingly.
Thank you for your comment. 
Good DIY.

Regards.

Yves

**Comment by tinue** [comment]
Thanks for your quick reply! Please allow me one more question: A2 and A3 are connected to Elevation and Throttle (E and T). How do these values mirror the LED of the copter?

I was assuming that the Arduino must be connected to the LED port of the Naza. This is a 4 pin port, which carries not only the LED signal, but also handles the USB protocol for updates. Therefore I was looking for some complex logic to "separate" the LED signal from the USB signal. But I can't find this.
So what I am wondering is this: How do you get the input for the Naza LED signal? How do you know that the LED flashes "green", and that you therefore have to flash the ring "green" also?

**Comment by diogeneyves** [comment]
You just tell the Arduino is connected to the 4-pin port, it’s the usb port, but it is shared for the activity of LEDs. The TX and RX changes function when the USB port is not used. It’s easy to see with an oscilloscope, even with a conventional measuring device. Nothing to do with to Elevation and Throttle. 
Positive (A) isn’t used because the output current is not enough to feed properly the Arduino.
I hope this is clearer now.

Regards

Yves

**Comment by tinue** [comment]
Thanks again! Since the Naza has pins labeled "T" and "E", I thought you meant these. But I understand now that T and E are actually the two LED connector signal pins. I think I almost understand everything now ;)

How do you drive the Arduino and the Pixel LEDs? Is the BEC from the Naza strong enough (PMU v2), or did you add your own BEC to the setup?

And one last question: What channel do you use to select the animation? I would think X1 or X2.

**Comment by tinue** [comment]
Ah, I found the source of the confusion by looking at the Naza itself... The "AETR" labels are for the four ports underneath the LED port, not for the pins of the LED port. So I would recommend to change your wire diagram.

**Comment by diogeneyves** [comment]
Yes, I drive the Arduino and the Pixel LEDs with a BEC by security.
And I use an extra channel of the receiver directly ( at choice),to select the animation.

**Comment by ASW27B** [comment]
Yves - you mentioned last year that you might make a kit for non-programmers. Did that ever happen? I'd like to get one with one LED ring for my hex.

Andy

**Comment by diogeneyves** [comment]
Hello,
 Yes it is still my intention. But I have a lot of professional work at the moment. 

I hope before the end of the year to put a kit on the market. 

Kind regards.

Yves

**Comment by mk3d** [comment]
Super job! Bravo et merci pour le partage d'info :+1: 

J'ai juste un peu de la peine avec ton schema de connexion.
Pourrais-tu éclairer ma lanterne?
Je ne pige pas comment tu connecte la LED à l'arduino, en in et out.

D'avance un grand merci et encore bravo pour le boulot!
